custom_components/heatmiserneo/switch.py

```diff
@@ -388,4 +408,122 @@ async def async_turn_off(self, **kwargs):
         self.data.standby = False
         self.async_write_ha_state()
-        
+
+
+class NeoTimerEntity(CoordinatorEntity, SwitchEntity):
+    """Represents a Heatmiser neoStat thermostat acting in TimeClock mode."""
+
+    def __init__(
+        self, neostat: NeoStat, coordinator: DataUpdateCoordinator, hub: NeoHub
+    ):
+        super().__init__(coordinator)
+        _LOGGER.debug(
+            f"Creating {type(self).__name__} for {neostat.name} {neostat.device_id}"
+        )
+
+        self._neostat = neostat
+        self._coordinator = coordinator
+        self._hub = hub
+        # Needed to filter the switches in the service call entity selector. This was the least worse option
+        self._attr_supported_features = SirenEntityFeature.DURATION
+
+    @property
+    def data(self):
+        """Helper to get the data for the current thermostat."""
+        (devices, _) = self._coordinator.data
+        neo_devices = {device.name: device for device in devices["neo_devices"]}
+        return neo_devices[self.name]
+
+    @property
+    def available(self):
+        """Return true if the entity is available."""
+        if self.data.offline:
+            return False
+        return True
+
+    @property
+    def device_class(self):
+        return SwitchDeviceClass.SWITCH
+
+    @property
+    def device_info(self):
+        return {
+            "identifiers": {
+                (
+                    DOMAIN,
+                    f"{self._coordinator.serial_number}_{self._neostat.serial_number}",
+                )
+            },
+            "name": self._neostat.name,
+            "manufacturer": "Heatmiser",
+            "model": f"{HEATMISER_PRODUCT_LIST[self.data.device_type]}",
+            "serial_number": self._neostat.serial_number,
+            "suggested_area": self._neostat.name,
+            "sw_version": self.data.stat_version,
+            "via_device": (DOMAIN, self._coordinator.serial_number),
+        }
+
+    @property
+    def extra_state_attributes(self):
+        """Return the additional state attributes."""
+        return {
+            "device_id": self._neostat.device_id,
+            "device_type": self._neostat.device_type,
+            "offline": self.data.offline,
+            "hold_on": self.data.hold_on,
+            "hold_time": ":".join(str(self.data.hold_time).split(":")[:2]),
+        }
+
+    @property
+    def name(self):
+        """Returns the name."""
+        return self._neostat.name
+
+    @property
+    def is_on(self):
+        """Return is_on status."""
+        return self.data.timer_on
+
+    async def async_turn_on(self, **kwargs):
+        """Turn on Timer"""
+        hold_minutes = 30
+        if ATTR_HOLD_DURATION in kwargs:
+            hold_duration = kwargs.get(ATTR_HOLD_DURATION)
+            hold_minutes = int(hold_duration.total_seconds() / 60)
+
+        _LOGGER.info(
+            f"Device ID: {self._neostat.device_id} - {self._neostat.name}: Executing turn_on() with:"
```

Comment:
Probably worth considering returning the unique ID here. Will help if the logs are across hubs for example.

Reply:
Changed device_id to unique_id